### PR TITLE
Make source required

### PR DESCRIPTION
**Pull request by kriscon-db**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [ ] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

Source was erroneously left off of the required list for CreateModelVersion when checked in.  In this PR we:

- Add source as required at at the API
- Add service handler checks for source existence
- Add an explicit test in the unit test to validate that source must be filled in.

### CODE REVIEW

File: server/src/main/java/io/unitycatalog/server/service/ModelService.java [hunk 1]
```diff
@@ -137,6 +137,10 @@ public HttpResponse createModelVersion(@AuthorizeKeys({
           @AuthorizeKey(value = REGISTERED_MODEL, key = "model_name")})
```
Comment (kriscon-db) [inline server/src/main/java/io/unitycatalog/server/service/ModelService.java:137]: @creechy With source being moved to required, I don't need to alter these annotations at all right?  This are purely for ACL checks on the actual securable keys?

Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/service/ModelService.java:137]: Yeah you shouldn't need to alter the annotations at all unless you want to include source in the ACL checking somehow.

Comment (kriscon-db) [reply server/src/main/java/io/unitycatalog/server/service/ModelService.java:137]: Nah.  I just wanted to double check on this.

**Review by tdas — APPROVED** [review]

**Review by rameshchandra — COMMENTED** [review]
LGTM

**Pull request merged**